src/main/java/net/neoforged/neoforge/common/extensions/IItemStackExtension.java

```diff
@@ -395,6 +397,13 @@ default <T> T getCapability(ItemCapability<T, @Nullable Void> capability) {
         return capability.getCapability(self(), null);
     }
 
+    /**
+     * @return An immutable {@link ResourceStack} with the current count.
```

Comment:
```suggestion
     * {@return an immutable {@link ResourceStack} with the current count}
```
to also generate a nice javadoc summary for the method.

Reply:
Can we compromise on these with us just doing the full manual handling for now?
```java
/**
 * Returns an immutable {@link ResourceStack} with the current count.
 * @return An immutable {@link ResourceStack} with the current count
 */
```

The reason for this compromise, is notably, IDEA has a bug that causes return tags to essentially [omit all inner in lined tags](https://youtrack.jetbrains.com/issue/IDEA-329877/Javadoc-does-not-render-correctly-for-inline-return). This is something they are working on (last I can see is March). While this wouldn't be too much of an issue, since it is expected many of neo users will be using IDEA, having it present in code as something that is functionally broken is likely not our best route.

This for example:
![image](https://github.com/user-attachments/assets/b6229c1c-b9bf-4ba8-8be4-3cb7f37c25f7)

In IDEA results into either the rendered doc being broken, or the java doc modal being broken.

![image](https://github.com/user-attachments/assets/5b3105ae-e117-4fab-96c9-8dbda65be722)
![image](https://github.com/user-attachments/assets/286af268-b721-49b8-8755-02ac18a3ac72)

Reply:
This is already used extensively in Neo. Just accept the suggestion.

Reply:
I am on the fence on this, MCF banned this style for that exact reason. NF never talked about it and it started slipping it. But @Soaryn has a point. We should write documentation that is not broken and that properly works and allows people to navigate freely through our code.

On the contrary it is supposed to be fixed sometime in the future, so why wait.

Unsure.

Reply:
Honestly, if we had a timeline on a solution from them I'd just bite the bullet and do it, but the whole point of documenting something to present the user with clear instructions or advise on usage, which as it stands now, it can't with other in lined tags. 

Since this has been a reported problem and the latest update on the matter I can find is from [March](https://github.com/JetBrains/intellij-community/pull/2671) I'd assume we are still likely a ways out from it being addressed. In the case we have other tags being used in the return statement, it is likely best for now to use the more verbose styling as perhaps unwanted as it is. HOWEVER, in the case of something that is purely just text in the return with no other tags, I agree we can make use of the return inline tag.

Reply:
In any case, using exclusively the (usual) `@return` tag should be avoided because it prevents any summary from being rendered. I am fine with just having `Returns whatever.` in the javadoc, i.e. no `@return` and no `{@return}` tag.

Reply:
The use of inline return tags was not banned under Forge, and Neo uses it extensively. A simple search for `{@return` on Neo's GitHub returns over 170 files and many of them have more than one occurrence, so it has hundreds of usages. In any case I maintain my stance from the previous PR thread:
> Regardless, we should not write our documentation based on IDE bugs which ought to be fixed at any time.

Once this bug is fixed by IntelliJ (as far as I'm aware it is not present on Eclipse, it's caused by how IntelliJ parses the file and turns the tokens into the AST structure), do we expect to modify documentation we've introduced to be rewritten to follow Java's guidance on using inline return tags for these cases?

Reply:
I wouldn't necessarily expect rewrites to happen, but functionally the verbose method is what that tag generates so it would only need to be rewritten if the documentation needs to actually change. If I understand correctly Oracle [advises](https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#@return) when using redundant messages in our case the body and the return are the same, to go ahead and do so.

Reply:
The article you've linked is so old it doesn't even mention the existence of inline return tags - the last java version it seems to mention is java 1.2. The inline return tag was created specifically to reduce the amount of redundant information.

Reply:
Seeing that the returns have been copied as summaries as well I am resolving this.